lib/promscrape/discovery/eureka/api.go

```diff
@@ -0,0 +1,81 @@
+package eureka
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promauth"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
+)
+
+var configMap = discoveryutils.NewConfigMap()
+
+type apiConfig struct {
+	client *discoveryutils.Client
+}
+
+func newAPIConfig(sdc *SDConfig, baseDir string) (*apiConfig, error) {
+	var (
+		ba    *promauth.BasicAuthConfig
+		token string
+	)
+	if sdc.Token != nil {
+		token = *sdc.Token
+	}
+	port := 80
+	if sdc.Port == nil {
+		sdc.Port = &port
```

Comment:
`sdc` fields shouldn't be modified here in order to possible prevent race condition when `newAPIConfig` is called from concurrently running goroutines for the same `sdc`.